crates/biome_js_analyze/src/lint/nursery/use_explicit_function_return_type.rs

```diff
@@ -214,3 +227,28 @@ fn is_direct_const_assertion_in_arrow_functions(func: &AnyJsFunction) -> bool {
 
     ts_ref.text() == "const"
 }
+
+/**
+ * Checks if a function is not part of a declaration
+ * JS_CALL_ARGUMENT_LIST
+ * - window.addEventListener('click', () => {});
+ * - const foo = arr.map(i => i * i);
+ * - setTimeout(function() { console.log("Hello!"); }, 1000);
+ *
+ * JS_ARRAY_ELEMENT_LIST
+ * - [function () {}, () => {}];
+ *
+ * JS_PARENTHESIZED_EXPRESSION
+ * - (function () {});
+ * - (() => {})();
+ */
+fn is_allow_expressions(func: &AnyJsFunction) -> bool {
```

Comment:
Can you rename the function in something more descriptive? This function only checks for a parenthesized expression, so it's misleading